torrent.go

```diff
@@ -1,13 +1,146 @@
+	Seeders     int             `json:"seeders"`
+	Leechers    int             `json:"leechers"`
+	User        string          `json:"username"`
+	Added       time.Time       `json:"added"`
+	Description string          `json:"descripiton"`
+	InfoHash    string          `json:"info_hash"`
+	ImdbID      string          `json:"imdb"`
+	// Magnet      string          `json:"magnet"`
+}
+
+var magnetTemplate *template.Template
+
+func init() {
+	magnetTemplate = template.Must(template.New("magnet").Parse(magnetTemplateText))
+}
+
+const magnetTemplateText = `magnet:?xt=urn:btih:{{.InfoHash}}&dn={{.Name}}{{range .Trackers}}&tr={{.}}{{end}}`
+
+var trackers = []string{
+	"udp://tracker.coppersurfer.tk:6969/announce",
+	"udp://9.rarbg.to:2920/announce",
+	"udp://tracker.opentrackr.org:1337",
+	"udp://tracker.internetwarriors.net:1337/announce",
+	"udp://tracker.leechers-paradise.org:6969/announce",
+	"udp://tracker.coppersurfer.tk:6969/announce",
+	"udp://tracker.pirateparty.gr:6969/announce",
+	"udp://tracker.cyberia.is:6969/announce",
+}
+
+// GetMagnet returns a Magnet for a torrent
+func (t *Torrent) GetMagnet() string {
+	var tpl bytes.Buffer
+	tplStruct := struct {
+		*Torrent
+		Trackers []string
+	}{
+		Torrent:  t,
+		Trackers: trackers,
+	}
+	err := magnetTemplate.Execute(&tpl, tplStruct)
+	if err != nil {
+		return ""
+	}
+	return tpl.String()
+}
+
+// UnmarshalJSON is a custom unmarshal function to handle timestamps and
+// boolean as int and convert them to the right type.
+func (t *Torrent) UnmarshalJSON(data []byte) error {
+	var aux struct {
+		ID          flexInt    `json:"id"`
+		Category    flexInt    `json:"category"`
+		Status      string     `json:"status"`
+		Name        string     `json:"name"`
+		NumFiles    flexInt    `json:"num_files"`
+		InfoHash    string     `json:"info_hash"`
+		Description string     `json:"descr"`
+		Leechers    flexInt    `json:"leechers"`
+		Seeders     flexInt    `json:"seeders"`
+		User        string     `json:"username"`
+		Size        flexInt    `json:"size"`
+		Added       flexInt    `json:"added"`
+		ImdbID      flexString `json:"imdb"`
+	}
+
+	// Decode json into the aux struct
+	if err := json.Unmarshal(data, &aux); err != nil {
+		return err
+	}
+
+	t.ID = int(aux.ID)
+	t.Category = TorrentCategory(int(aux.Category))
+	t.Status = UserStatus(aux.Status)
+	t.Name = aux.Name
+	t.NumFiles = int(aux.NumFiles)
+	t.Size = uint64(aux.Size)
+	t.Seeders = int(aux.Seeders)
+	t.Leechers = int(aux.Leechers)
+	t.User = aux.User
+	t.Added = time.Unix(int64(aux.Added), 0)
+	t.Description = aux.Description
+	t.InfoHash = aux.InfoHash
+	t.ImdbID = string(aux.ImdbID)
+	// t.Magnet = t.GetMagnet()
+
+	return nil
+}
+
+// flexInt represents a int that is a string or a int in json
+// - if it's a int, it represents the int
+// - if it's a string, it represents the string converted in int
+type flexInt int
+
+func (fi *flexInt) UnmarshalJSON(b []byte) error {
+	if b == nil || len(b) == 0 {
```

Comment:
```go
if len(b) == 0 {
```
is enough: https://play.golang.org/p/07tDrxA9qgc

Reply:
Absolutely right, I usually spot this with staticcheck but didn't do it yet on the PR, I juste did and found other issues